www/js/lib/zimArchive.js

```diff
@@ -91,17 +92,13 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8', 'uiUtil'],
         } else {
             if (/.*zim..$/.test(path)) {
                 // split archive
-                that._searchArchiveParts(storage, path.slice(0, -2)).then(function(fileArray) {
+                that._searchArchiveParts(storage, path.slice(0, -2)).then(function (fileArray) {
                     createZimfile(fileArray);
-                }, function(error) {
-                    uiUtil.systemAlert("Error reading files in split archive " + path + ": " + error, "Error reading archive files");
-                });
+                }).catch(callbackError("Error reading files in split archive " + path + ": " + error, "Error reading archive files"));
```

Comment:
I'm not sure this will pass the error variable. I may be wrong, but normally you have to do it with an anonymous function like this:
```
}).catch(function (error) {
    callbackError("Error reading files in split archive " + path + ": " + error, "Error reading archive files");
});
```

Reply:
Yes, right, my bad